osu.Game/Rulesets/Objects/SliderPath.cs

```diff
@@ -35,35 +27,24 @@ public struct SliderPath : IEquatable<SliderPath>
         /// <summary>
         /// Creates a new <see cref="SliderPath"/>.
         /// </summary>
-        /// <param name="type">The type of path.</param>
-        /// <param name="controlPoints">The control points of the path.</param>
-        /// <param name="expectedDistance">A user-set distance of the path that may be shorter or longer than the true distance between all
-        /// <paramref name="controlPoints"/>. The path will be shortened/lengthened to match this length.
-        /// If null, the path will use the true distance between all <paramref name="controlPoints"/>.</param>
+        /// <param name="segments">The path segments.</param>
+        /// <param name="expectedDistance">A user-set distance of the path that may be shorter or longer than the true distance. The path will be shortened/lengthened to match this length.
+        /// If null, the path will use the true distance.</param>
         [JsonConstructor]
-        public SliderPath(PathType type, Vector2[] controlPoints, double? expectedDistance = null)
+        public SliderPath(PathSegment[] segments, double? expectedDistance = null)
```

Comment:
can we add a fallback constructor for single segment paths rather than complicate all the existing usages with single item array logic?